File: src/main/java/com/kmarzecki/communicator/util/DateProviderImpl.java
package com.kmarzecki.communicator.util;

import org.springframework.stereotype.Component;

import java.time.LocalDateTime;
import java.util.Date;


@Component
class DateProviderImpl implements DateTimeProvider {
    @Override
    public Date currentDate() {
        return new Date();
    }

    @Override
    public LocalDateTime currentLocalDateTime() {
        return LocalDateTime.now();
    }
}

File: src/main/kotlin/com/kmarzecki/trollo/util/DateProviderImpl.kt
package com.kmarzecki.trollo.util

import org.springframework.stereotype.Component
import java.time.LocalDateTime
import java.util.*

@Component
internal class DateProviderImpl : DateTimeProvider {
    override fun currentDate(): Date {
        return Date()
    }

    override fun currentLocalDateTime(): LocalDateTime? {
        return LocalDateTime.now()
    }
}